rust/connlib/tunnel/src/device_channel/tun_linux.rs

```diff
@@ -204,27 +166,24 @@ impl Tun {
                 .header
                 .index;
 
-            let message = match route {
-                IpNetwork::V4(ipnet) => make_route_v4(index, &handle, ipnet).message_mut().clone(),
-                IpNetwork::V6(ipnet) => make_route_v6(index, &handle, ipnet).message_mut().clone(),
-            };
+            for route in routes.iter().filter(|r| !current_routes.contains(r)) {
+                add_route(route, index, &handle).await;
+            }
 
-            match handle.route().del(message).execute().await {
-                Ok(_) => Ok(()),
-                Err(err) => {
-                    tracing::error!(%route, "failed to add route: {err:#?}");
-                    Ok(())
-                }
+            for route in current_routes.iter().filter(|r| !routes.contains(r)) {
+                delete_route(route, index, &handle).await;
             }
```

Comment:
If you use a `HashSet`, getting the changes is as simple as calling `difference` on it.